### PR TITLE
fix: Trim hyphen when slugifying team name

### PR DESCRIPTION
**Pull request by saikarsis**
## What does this PR do?

Improves the auto-generated slug when creating a team by trimming unexpected trailing hyphens.
Unexpected trailing hyphens occur when inputting a team name with punctuation, such as "My team!"

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Start creating a new team and input the name "My team!" and observe the absence of a trailing hyphen in the slug.
- [ ] Start creating a new team and input the name "Another team-" and observe the retention of a trailing hyphen in the slug.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/features/ee/teams/components/CreateANewTeamForm.tsx [hunk 1]
```diff
@@ -16,6 +16,19 @@ const querySchema = z.object({
   returnTo: z.string(),
 });
 
+function noTrailingHyphenSlugify(str: string) {
+  const slugified_str = slugify(str);
```
Comment (hariombalhara) [inline packages/features/ee/teams/components/CreateANewTeamForm.tsx:20]: I am not sure what testcase I missed but the objective can be met by 
```javascript
function noTrailingHyphenSlugify2(str) {
  const slugified_str = slugify(str);
  return slugified_str.replace(/-*$/, '')
}`

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Thanks for your contribution @saikarsis 🙏 I have left a suggestion that can possible improve the code. Please check that.

Blocking merge till that time.

File: packages/features/ee/teams/components/CreateANewTeamForm.tsx (hunk 1 above)
Comment (hariombalhara) [reply packages/features/ee/teams/components/CreateANewTeamForm.tsx:20]: Also, I think we can improve the slugify logic itself to tackle this. No one would want trailing hyphens.

**Comment by hariombalhara** [comment]
Moving it to draft till requested comments are addressed.

**Comment by PeerRich** [comment]
@hariombalhara what do we do here?

**Pull request closed without merging**